main/src/hadd.cxx

```diff
@@ -1,39 +1,47 @@
 /**
   \file hadd.cxx
-  \brief This program will add histograms (see note) and Trees from a list of root files and write them to a target root file.
-  The target file is newly created and must not be
-  identical to one of the source files.
+  \brief This program will add histograms (see note) and Trees from a list of root files and write them to a target root
+  file. The target file is newly created and must not be identical to one of the source files.
 
   Syntax:
   ```{.cpp}
-       hadd targetfile source1 source2 ...
+       hadd [flags] targetfile source1 source2 ...
   ```
-  or
-  ```{.cpp}
-       hadd -f targetfile source1 source2 ...
-  ```
-  (targetfile is overwritten if it exists)
+
+  Flags can be passed before or after the positional arguments.
+  The first positional (non-flag) argument will be interpreted as the targetfile.
+  After that, the first sequence of positional arguments will be interpreted as the input files.
+  If two sequences of positional arguments are separated by flags, all sequences but the first
+  will be ignored.
```

Comment:
> all sequences but the first  will be ignored.

If we are changing behaviour in this PR it might be worth being a bit cleaner about this and warn or raise an error if this situation arises.